.github/workflows/test-and-build.yml

```diff
@@ -0,0 +1,69 @@
+name: CI Pipeline
+
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+  release:
+    types:
+      - published
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set up JDK 17
```

Comment:
you should step up to 21

Reply:
For now, I will keep JDK 17 to maintain consistency across the project. I plan to consider an upgrade to JDK 21 in the future.